src/main/kotlin/no/nav/familie/ba/sak/behandling/vilkår/VilkårsvurderingUtils.kt

```diff
@@ -41,33 +41,53 @@ object VilkårsvurderingUtils {
      * Funksjon som tar inn endret vilkår og muterer person resultatet til å få plass til den endrede perioden.
      */
     fun muterPersonResultatPut(personResultat: PersonResultat, restVilkårResultat: RestVilkårResultat) {
+        validerAvslagUtenPeriodeMedLøpende(personResultat, restVilkårResultat)
         val kopiAvVilkårResultater = personResultat.vilkårResultater.toList()
 
-        kopiAvVilkårResultater.forEach {
-            tilpassVilkårForEndretVilkår(
-                    personResultat = personResultat,
-                    vilkårResultat = it,
-                    restVilkårResultat = restVilkårResultat
-            )
+        kopiAvVilkårResultater
+                .filter { !it.erAvslagUtenPeriode() }
+                .forEach {
+                    tilpassVilkårForEndretVilkår(
+                            personResultat = personResultat,
+                            vilkårResultat = it,
+                            restVilkårResultat = restVilkårResultat
+                    )
+                }
+    }
+
+    fun validerAvslagUtenPeriodeMedLøpende(personSomEndres: PersonResultat, vilkårSomEndres: RestVilkårResultat) {
+        val resultaterPåVilkår = personSomEndres.vilkårResultater.filter { it.vilkårType == vilkårSomEndres.vilkårType }
+        when {
+            vilkårSomEndres.erAvslagUtenPeriode() && resultaterPåVilkår.any { it.resultat == Resultat.OPPFYLT && it.harFremtidigTom() } -> throw FunksjonellFeil(
+                    "Finnes løpende oppfylt ved forsøk på å legge til avslag uten periode ",
+                    "Du kan ikke legge til avslag uten datoer fordi det finnes oppfylt løpende periode på vilkåret.")
+            vilkårSomEndres.harFremtidigTom() && resultaterPåVilkår.any { it.erAvslagUtenPeriode() } -> throw FunksjonellFeil(
+                    "Finnes avslag uten periode ved forsøk på å legge til løpende oppfylt",
+                    "Du kan ikke legge til løpende periode fordi det er vurdert avslag uten datoer på vilkåret.")
```

Comment:
```suggestion
            vilkårSomEndres.erAvslagUtenPeriode() && resultaterPåVilkår.any { it.resultat == Resultat.OPPFYLT && it.harFremtidigTom() } -> 
            throw FunksjonellFeil(
                    "Finnes løpende oppfylt ved forsøk på å legge til avslag uten periode ",
                    "Du kan ikke legge til avslag uten datoer fordi det finnes oppfylt løpende periode på vilkåret.")
            vilkårSomEndres.harFremtidigTom() && resultaterPåVilkår.any { it.erAvslagUtenPeriode() } -> 
            throw FunksjonellFeil(
                    "Finnes avslag uten periode ved forsøk på å legge til løpende oppfylt",
                    "Du kan ikke legge til løpende periode fordi det er vurdert avslag uten datoer på vilkåret.")
```

Er dette mer lesbart kanskje?